form/src/App.js

```diff
@@ -13,6 +13,23 @@ function App() {
   const [isOpen, setIsOpen] = React.useState(false);
   const [content, setContent] = React.useState("Transitioning...");
   const [formData, setFormData] = React.useState({});
+  const [url, setUrl] = React.useState("");
+  const [filename, setFilename] = React.useState("");
+
+  useEffect(() => {
+    var data = new Blob([content]);
+    setUrl(URL.createObjectURL(data));
+    var {vendor, model, variant} = formData;
+    if((vendor !== undefined && model !== undefined) || variant !== undefined){
+    var name = vendor.toLowerCase()+'_'+ model.toLowerCase();
+    if(variant !== undefined){
+      name = name +'-'+ variant + ".yaml";
```

Comment:
the `yaml` suffix is only added if a variant is set, it should be added in any case.